### PR TITLE
Add cargo clippy lints to the CI

### PR DESCRIPTION
**Pull request by hug-dev**
This commit fixes the clippy cargo group lints and adds the check on the
CI script.
Also fixes some lints in the build script.

See #91 

Signed-off-by: Hugues de Valon <hugues.devalon@arm.com>

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -17,15 +17,12 @@
     const_err,
     dead_code,
     improper_ctypes,
-    legacy_directory_ownership,
     non_shorthand_field_patterns,
     no_mangle_generic_items,
     overflowing_literals,
     path_statements,
     patterns_in_fns_without_body,
-    plugin_as_library,
```
Comment (ionut-arm) [inline src/lib.rs]: What's with these ones?

File: build.rs [hunk 2]
```diff
@@ -12,6 +12,36 @@
 // WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+#![deny(
```
Comment (ionut-arm) [inline build.rs]: No way to just have these in a "global" file somewhere, something like a `./clippy.toml`? :(

File: build.rs [hunk 3]
```diff
@@ -75,63 +105,67 @@ fn get_value_from_table<'a>(table: &'a Value, key: &str) -> Result<&'a Value> {
...
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ARM64_MBED_COMPILER.to_string()),
+                toolchain
+                    .mbed_archiver
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ARM64_MBED_ARCHIVER.to_string()),
+            )
+        } else {
+            let toolchain;
+            toolchain = mbed_config.native.as_ref().ok_or_else(|| {
```
Comment (ionut-arm) [inline build.rs]: Do you think we should keep the subtable as required but the keys in it as optional? The other way around sounds more sensible - put a subtable in your `build-config.toml` with the correct keys if you want to change smth

File: src/lib.rs (hunk 1 above)
Comment (hug-dev) [reply src/lib.rs]: They have been removed as rustc lints and are hard errors now. When adding them there is a warning.

File: build.rs (hunk 2 above)
Comment (hug-dev) [reply build.rs]: I don't think it is available at the moment, the `clippy.toml` file seems to be for [configuring existing lints](https://github.com/rust-lang/rust-clippy#configuration).

File: build.rs (hunk 3 above)
Comment (hug-dev) [reply build.rs]: I don't really mind, maybe that could be discussed as part of #62 ?

Comment (ionut-arm) [reply build.rs]: Sure!

**Pull request merged**